### PR TITLE
PR: Add initial `pre-commit` config file and fix CI (macOS image label and codecov token errors)

### LINKED ISSUE #260: Add `pre-commit` setup for formatting
* Setup `pre-commit.ci`: https://[pre-commit.ci](https://pre-commit.ci/)/
* Use `ruff` and `ruff-format`: https://[pre-commit](https://pre-commit.com/hooks.html).com/hooks.html (search for `ruff`)

### PR DESCRIPTION
**Pull request by dalthviz**
Fixes #260

### CODE REVIEW

**Comment by dalthviz** [comment]
@ccordoba12 is there anything missing with this or should I go ahead and merge it?

**Comment by ccordoba12** [comment]
Sorry @dalthviz, I forgot to review your PR when you submit it. It looks good to me at first glance.

My main recommendation is to add the commit hash `ec7f2c8ecdcb9a7f4b7e451536f4aab3fa251916` to `.git-blame-ignore-revs` so that it's ignored by `git blame`.

File: .github/workflows/macos-tests.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
   macos:
     name: Mac Py${{ matrix.PYTHON_VERSION }} - ${{ matrix.QT_BINDING }}
     timeout-minutes: 15
-    runs-on: macos-latest
+    runs-on: macos-12
```
Comment (ccordoba12) [inline .github/workflows/macos-tests.yml:15]:
```suggestion
    runs-on: macos-13
```

macOS 12 is going to removed from Github CIs in a few weeks.

**Review by ccordoba12 — COMMENTED** [review]
One additional tiny suggestion, the rest looks good to me.

**Review by ccordoba12 — APPROVED** [review]
Looks good to me now, thanks @dalthviz!

**Pull request merged**